libbeat/cmd/instance/ilm.go

```diff
@@ -0,0 +1,203 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package instance
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"encoding/json"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/outputs/elasticsearch"
+)
+
+type ilmConfig struct {
+	RolloverAlias string `config:"ilm.rollover_alias" `
+	Pattern       string `config:"ilm.pattern"`
+}
+
+var ILMPolicy = common.MapStr{
+	"policy": common.MapStr{
+		"phases": common.MapStr{
+			"hot": common.MapStr{
+				"actions": common.MapStr{
+					"rollover": common.MapStr{
+						"max_size": "50gb",
+						"max_age":  "30d",
+					},
+				},
+			},
+		},
+	},
+}
+
+const (
+	ILMPolicyName     = "beats-default-policy"
+	ILMDefaultPattern = "%7Bnow%2Fd%7D-000001" // {now/d} escaped
+)
+
+// Build and return a callback to load ILM write alias
+func (b *Beat) writeAliasLoadingCallback() (func(esClient *elasticsearch.Client) error, error) {
+	callback := func(esClient *elasticsearch.Client) error {
+		if b.Config.ILM == nil {
+			b.Config.ILM = common.NewConfig()
+		}
+
+		config, err := getILMConfig(b)
+		if err != nil {
+			return err
+		}
+
+		// This always assume it's a date pattern by sourrounding it by <...>
+		firstIndex := fmt.Sprintf("%%3C%s-%s%%3E", config.RolloverAlias, config.Pattern)
+
+		// Check if alias already exists
+		status, b, err := esClient.Request("HEAD", "/_alias/"+config.RolloverAlias, "", nil, nil)
+		if err != nil && status != 404 {
+			logp.Err("Failed create write alias: %s: %+v", err, string(b))
```

Comment:
Wrong log message, this is not trying to `create` but to `check` for write alias.

Reply:
fixed